smart-contracts/contracts/Unlock.sol

```diff
@@ -642,6 +643,37 @@ contract Unlock is UnlockInitializable, UnlockOwnable {
     return globalTokenSymbol;
   }
 
-  // required to receive ETH
+  // for doc, see IUnlock.sol
+  function postLockUpgrade() public {
+    // check if lock hasnot already been deployed here and version is correct
+    if (
+      locks[msg.sender].deployed == false
+      && 
+      IPublicLock(msg.sender).publicLockVersion() == 13 
+      && 
+      IPublicLock(msg.sender).unlockProtocol() != address(this)
+    ) {
+      IUnlock previousUnlock = IUnlock(
+        IPublicLock(msg.sender).unlockProtocol()
+      );
+
+      (
+        bool deployed, 
+        uint totalSales, 
+        uint yieldedDiscountTokens
+      ) = previousUnlock.locks(msg.sender);
+
+      // record lock from old Unlock in this one
+      if (deployed) {
```

Comment:
Do we want to revert if this is called by a contract that is _not_ deployed on the previous lock?